client/server/pages/index.js

```diff
@@ -636,7 +636,7 @@ export default function pages() {
 					res.redirect(
 						'https://wordpress.com/wp-login.php?redirect_to=https%3A%2F%2Fwordpress.com%2Fplans'
 					);
-				} else {
+				} else if ( ! config.isEnabled( 'jetpack-cloud/connect' ) ) {
```

Comment:
I'm late to the party, but I found this bug when looking for a test file today: this new check creates an unhandled request if `req.context.isLoggedIn===false` and `config.isEnabled( 'jetpack-cloud/connect' )===true`. What should happen in that case?